File: dsa/binary-search/BinarySearch.kt
fun main() {
    print("Enter Sorted Array to Search: ")
    val nums = readln()
    print("Enter the No. to Search in Array: ")
    val target = readln().toInt()
    val pos = search(nums, target)
    if (pos == -1) print("$target Not Found")
    else print("$target is in $pos position")
}

private fun search(nums: String, target: Int): Int {
    val num = nums.split(" ").map {
        it.toInt()
    }.toIntArray()
    return binarySearch(num, target)
}

private fun binarySearch(nums: IntArray, target: Int): Int {
    var start = 0
    var end = nums.lastIndex
    var mid: Int
    while (start <= end) {
        mid = (start + end) / 2
        if (nums[mid] == target) return mid
        if (target > nums[mid]) {
            start = mid + 1
        } else {
            end = mid - 1
        }
    }
    return -1
}

File: dsa/binary-search/BinarySearch.c
#include <stdio.h>

int main()
{
    int n;
    scanf("%d",&n);
    int arr[n];
    for(int i=0;i<n;i++) scanf("%d",&arr[i]);
    int target;
    scanf("%d",&target);
    int start=0,end=n-1,mid,c=0;
    while(start<=end){
        mid=(start+end)/2;
        if(arr[mid]==target) {c++;break;}
        else if(arr[mid]<target) start=mid+1;
        else end=mid-1;
    }
    if(c>0) printf("%d",mid+1);
    else printf("Not found");
    return 0;
}
